src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +31,58 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfigWithTrustedCertificate(remoteCert *x509.Certificate) (*tls.Config, error) {
+	if remoteCert == nil {
+		return nil, fmt.Errorf("invalid remote public key")
+	}
+
+	config := &tls.Config{}
+
+	remoteCert.IsCA = true
+	config.RootCAs.AddCert(remoteCert)
+
+	// Always use public key DNS name rather than server cert, so that it matches.
+	if len(remoteCert.DNSNames) > 0 {
+		config.ServerName = remoteCert.DNSNames[0]
+	}
+
+	return config, nil
+}
+
+func GetRemoteCertificate(address string) (*x509.Certificate, error) {
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	_, _, err := net.SplitHostPort(address)
```

Comment:
super small nit: It makes sense to do the input validation at the beginning of the function. Otherwise you create the `httpClient` for no reason.